transforms/class.js

```diff
@@ -258,24 +258,34 @@ module.exports = (file, api, options) => {
       })
       .forEach(path => j(path).replaceWith(j.identifier('props')));
 
-  const inlineGetInitialState = getInitialState =>
-    getInitialState.value.body.body.map(statement => {
-      if (statement.type === 'ReturnStatement') {
-        return j.expressionStatement(
+  const inlineGetInitialState = getInitialState => {
+    const functionExpressionAST = j(getInitialState.value);
+
+    return functionExpressionAST
+      .find(j.ReturnStatement)
+      .forEach(path => {
+        j(path).replaceWith(j.expressionStatement(
           j.assignmentExpression(
             '=',
             j.memberExpression(
               j.thisExpression(),
               j.identifier('state'),
               false
             ),
-            statement.argument
+            path.value.argument
           )
-        );
-      }
-
-      return statement;
-    });
+        ));
+
+        if ( // FIXME is there a better way to check this?
```

Comment:
Do you ever want to omit `return` in any case other than when it is direct child of method body?